lib/tether/ops_linux.go

```diff
@@ -213,7 +213,7 @@ func assignIP(link netlink.Link, endpoint *metadata.NetworkEndpoint) (bool, erro
 	}
 
 	// Set IP address if it's specified - this is now named for for the network role rather than the nic
-	if len(endpoint.Static.IP) > 0 {
+	if endpoint.Static.IP != nil && !endpoint.Static.IP.IsUnspecified() {
```

Comment:
Dug into this - it's actually something I'd looked at at the beginning of #1127. It's related to handling in extraconfig that I've not hand time to fix. There's now a test being skipped to assert that unspecified structures remain nil.
I tried switching to IsUnspecified but in the end I had to go back the len(IP) > 0.